Enable Remotes for Action Commands

ikaneshiro opened the pull request:
**Description of the Pull Request (PR):**

Missed `library` uris in `action` commands in #2829. This fixes that by checking for a default remote and using that.

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)


Attn: @singularity-maintainers

ikaneshiro committed:
have library uri's for action commands use remote if set

bauerm97 reviewed (dismissed):
Actually wait, looks like you broke CI

mem commented on cmd/internal/cli/actions.go:
```diff
@@ -210,6 +211,32 @@ func setVM(cmd *cobra.Command) {
+
+	// if we can load config and if default endpoint is set, use that
+	// otherwise fall back on regular authtoken and URI behavior
+	e, err := sylabsRemote(remoteConfig)
```
could you please name the variable `endpoint` (or something else more descriptive if you prefer)? It's not immediately obvious from the function name what this value is supposed to be, which makes the rest of the code a little hard to read.

ikaneshiro replied:
Will do. I'll also change the other places in the CLI where I was just using `e`

mem commented on cmd/internal/cli/actions.go:
```diff
@@ -210,6 +211,32 @@ func setVM(cmd *cobra.Command) {
 
+// returns url for library and sets auth token based on remote config
+// defaults to https://library.sylabs.io
+func handleActionRemote(cmd *cobra.Command) (remoteURL string) {
```
Please no named returns. See [here](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters).

That takes care of the naked return on lines #224 and #237.

Also, because of the way this function works, it's probably a good idea to return both the remote URL and an error in case it cannot be obtained.

ikaneshiro replied:
Refactored this function a bit to get rid of the named and naked returns. I think it's ok to just warn or exit on an error here. I don't see a need for the caller to handle the error.

mem commented on cmd/internal/cli/actions.go:
```diff
@@ -210,6 +211,32 @@ func setVM(cmd *cobra.Command) {
+	}
+
+	authToken = e.Token
+	uri, err := e.GetServiceURI("library")
```
This is probably better written as:

```
remoteURL, err := e.GetServiceURI("library")
if err != nil {
    sylog.Warning("Unable to get library service URI: %v", err)
}
return remoteURL // or return remoteURL, err if you modify the return value as suggested
```

ikaneshiro replied:
This doesn't quite work since `GetServiceURI()` returns `""` in the case of an error and we want to always return a correct `URI` from this function. During the refactoring for the previous comment I think I address this.

ikaneshiro committed:
Merge branch 'master' into action_remotes

ikaneshiro committed:
Addressing marcelos comments

	Renamed endpoint variable from e to endpoint in multiple
	locations. Refactored handleActionRemote() to remove naked
	and named returns.

ikaneshiro committed:
Merge remote-tracking branch 'upstream/master' into action_remotes

mem reviewed (approved):
Thanks!

mem committed:
Merge branch 'master' into action_remotes
